### PR TITLE
8298298: NMT: count deltas are printed with 32-bit signed size

### PR DESCRIPTION
**Pull request by GoeLin**
A useful bugfix for NMT>

Trivial resolves in code.
The test that got extended by  a few new cases is not in 17. 
I added the test body so that the new test cases are checked.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] [JDK-8298298](https://bugs.openjdk.org/browse/JDK-8298298) needs maintainer approval
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298298](https://bugs.openjdk.org/browse/JDK-8298298): NMT: count deltas are printed with 32-bit signed size (**Bug** - P3 - Approved)

### Reviewers
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**) ⚠️ Review applies to [4c6b032d](https://git.openjdk.org/jdk17u-dev/pull/1824/files/4c6b032d235346b63fb758046be73bba9bf8c605)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to [4c6b032d](https://git.openjdk.org/jdk17u-dev/pull/1824/files/4c6b032d235346b63fb758046be73bba9bf8c605)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/1824/head:pull/1824` \
`$ git checkout pull/1824`

Update a local copy of the PR: \
`$ git checkout pull/1824` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/1824/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1824`

View PR using the GUI difftool: \
`$ git pr show -t 1824`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/1824.diff">https://git.openjdk.org/jdk17u-dev/pull/1824.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/1824#issuecomment-1745661011)

### CODE REVIEW

File: src/hotspot/share/utilities/globalDefinitions.hpp [hunk 1]
```diff
@@ -141,6 +141,7 @@ class oopDesc;
 #define INTPTR_FORMAT_W(width)   "%" #width PRIxPTR
 
 #define SSIZE_FORMAT             "%"   PRIdPTR
+#define SSIZE_PLUS_FORMAT        "%+"         PRIdPTR
```
Comment (RealLucy) [inline src/hotspot/share/utilities/globalDefinitions.hpp:144]: alignment would be nice.

**Review by RealLucy — APPROVED** [review]
LGTM.
One minimal formatting request. 
Thanks for the backport work.

**Review by tstuefe — APPROVED** [review]
Good!

(looking guiltily at my feet)

...Thomas

**Comment by GoeLin** [comment]
Thanks for implementing these in first place, Thomas!

**Pull request closed without merging**